### ISSUE TITLE
Danalock Connection

### ISSUE THREAD

**Issue by peaoo** [athombv/homey#412]
**What is your Homey version (Settings → System)?**

.29

**What did you try to achieve?**

When I try to connect my Danalock most of the time the Issue #240 occours. But sometimes I am able to connect it (It seems this happens when I have a reboot between removing node and adding it), but it has no options to open/close. In a flow there are only "set configuration parameter" or "reset configuration" choosable. After a few seconds the Danalock times out with "Error: Timeout waiting for report".
That is the code: 4EB8AA6EB8

**What was the result?**

Danalock timed out

**What did you expect as result?**

**Any other remarks**

Issue #240 is linked. But it seems to be more device specific.

**Comment by peaoo** [comment]
Nothing changed after .30

**Comment by peaoo** [comment]
I added D-Link Door/Window sensors without any problems.

**Comment by peaoo** [comment]
I took a few screens:
![danal](https://cloud.githubusercontent.com/assets/17235241/14608799/6ab9a27e-0587-11e6-9d6e-288dd150569c.jpg)

And I noticed that when I click on the grey lock icon the Danalock will be displayed as "Error: Timeout waiting for a report".

**Comment by peaoo** [comment]
@ocatie I know this is a very low priority to fix (I think I am the only Danalock/Homey User atm :D) but it would be nice to get a little bit of feedback. (maybe an ETA e.g.: after Z-Wave rewite)

**Comment by bvdbos** [comment]
Did you try with 0.8.35 or with 0.8.32 ? Setting parameters has been changed...

**Comment by peaoo** [comment]
I just tried it again, after I removed its node I am not able to add or remove nodes anymore. I am getting this error: "Error: Cannot read property 'addNode' of undefined". I rebootet 4-5 times but this error is still there. No matter if I try to add a device or if I try to remove a node. After this error appeared I am suffering to Issue #240 until I reboot and have the "Cannot read property" Error again.

**Comment by peaoo** [comment]
I just recorded a video, maybe this helps.
Didn't find anything else about this error.
https://youtu.be/f-_n-x9MefU

**Comment by bvdbos** [comment]
Cannot read propery addnode of undefined
looks like a very strange error to me... Looks like some zwave-core functions aren't loading?

**Comment by ocatie** [comment]
Could you please reboot, reproduce the error and send a system log?

**Comment by peaoo** [comment]
Unfortunately I didn't made an error report when I recorded the video. The error seemed to disappeared the next morning. 

I tried to add the Danalock again and it worked this time. (The difference was that "Interviewing Device" when adding actually finished this time and didn't run into an endless loop). I got the warning that it is added as non secure Z-Wave device and selected lock.
Unfortunately I am still having the issue that I only have 2 Flow cards (Reset Configuration and Set configuration).
A difference to the erarlier firmware is, that I have following screen at the Device overview:
![timeout](https://cloud.githubusercontent.com/assets/17235241/15775124/81b336b2-2980-11e6-8b8c-d8f23b0b97e9.png)

The lock is displayed as connected until I am coming to this screen. The screen stays there up to 5 seconds, followed by a spinning wheel. After that the lock becomes unaviable and is displayed as "Timeout waiting for report" for 30-60sec.

Log: 4F12925FF8
